Verify upcoming V4.0_2024_1119_06da8476 release

I just updated the AIC8800 vendor code to a new release, and fixed failed patches along the way. This is currently working on our Cubie A5E running Allwinner 5.15 kernel. My local build is available here:

[aic8800.zip](https://github.com/user-attachments/files/21015180/aic8800.zip)

@CodeChenL please test this with our Rockchip products (SDIO & USB, 5.10 & 6.1).

@amazingfate can you please check with the latest kernel on Armbian to make sure we won't have any regression?

DKMS error with rock-5c_bookworm_kde_t1
```
rock@rock-5c:~$ cat /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/make.log
DKMS make.log for aic8800-usb-3.0+git20240327.3561b08f-7 for kernel 6.1.84-2-rk2410 (aarch64)
Fri Jul  4 10:55:30 UTC 2025
make: Entering directory '/usr/src/linux-headers-6.1.84-2-rk2410'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: aarch64-linux-gnu-gcc (Debian 10.2.1-6) 10.2.1 20210110
  You are using:           gcc (Debian 12.2.0-14) 12.2.0
make[1]: warning: -j8 forced in makefile: resetting jobserver mode.
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aic_bluetooth_main.o
make[2]: warning: -j8 forced in makefile: resetting jobserver mode.
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aicbluetooth.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_wakelock.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_msg_tx.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aicwf_usb.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_msg_rx.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aic_txrxif.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_utils.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aicbluetooth_cmds.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_cmds.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_irqs.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aic_compat_8800d80.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aic_compat_8800d80x2.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_cfgfile.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_strs.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/md5.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aicwf_rx_prealloc.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aicwf_txq_prealloc.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_tx.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_txq.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mesh.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_platform.o
  CC [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_pci.o
  LD [M]  /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic_load_fw/aic_load_fw.o
/var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c: In function ‘rwnx_set_vht_capa’:
/var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c:936:5: error: this ‘if’ clause does not guard... [-Werror=misleading-indentation]
  936 |     if(rwnx_hw->usbdev->chipid <= PRODUCT_ID_AIC8800D81)
      |     ^~
/var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c:939:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  939 |         band_2GHz->vht_cap.vht_supported = true;
      |         ^~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [scripts/Makefile.build:250: /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [scripts/Makefile.build:500: /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800/aic8800_fdrv] Error 2
make: *** [Makefile:2015: /var/lib/dkms/aic8800-usb/3.0+git20240327.3561b08f-7/build/USB/driver_fw/drivers/aic8800] Error 2
make: Leaving directory '/usr/src/linux-headers-6.1.84-2-rk2410'
rock@rock-5c:~$ 
```

--- a/debian/patches/fix-usb-build.patch
+++ b/debian/patches/fix-usb-build.patch
@@ -1,0 +1,25 @@
+diff --git a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile
+index d0970cb..656af12 100644
+--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile
++++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile
+@@ -1,6 +1,6 @@
+ EXTRA_CFLAGS += $(USER_EXTRA_CFLAGS)
+ EXTRA_CFLAGS += -Wno-implicit-fallthrough
+-#EXTRA_CFLAGS += -Wno-unused-variable
++EXTRA_CFLAGS += -Wno-unused-variable
+ 
+ RWNX_VERS_NUM := 6.4.3.0
+ 
+diff --git a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c
+index f78ca5f..3e7891f 100644
+--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c
++++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c
+@@ -936,7 +936,7 @@ static void rwnx_set_vht_capa(struct rwnx_hw *rwnx_hw, struct wiphy *wiphy)
+     if(rwnx_hw->usbdev->chipid <= PRODUCT_ID_AIC8800D81)
+         nss = 1;
+ 
+-	band_2GHz->vht_cap.vht_supported = true;
++    band_2GHz->vht_cap.vht_supported = true;
+ 		if (rwnx_hw->mod_params->sgi80)
+ 			band_2GHz->vht_cap.cap |= IEEE80211_VHT_CAP_SHORT_GI_80;
+ 		if (rwnx_hw->mod_params->stbc_on)

--- a/debian/patches/series
+++ b/debian/patches/series
@@ -16,3 +16,4 @@ fix-linux-6.15-build.patch
 fix-aic_btusb-implicit-declare-compat_ptr.patch
 fix-allwinner-dkms.patch
 fix-linux-6.16-build.patch
+fix-usb-build.patch

